package/gluon-mesh-vpn-wireguard/files/usr/bin/wg-broker

```diff
@@ -0,0 +1,62 @@
+#!/bin/sh
+run_broker() {
+  local interface="$1"
+  local pubkey="$2"
+  local port
+  local interval=5
+
+  group_peers=$(uci show wireguard|grep mesh_vpn_backbone_peer|grep =peer|cut -d"." -f2|cut -d= -f1)
+
+  for i in $group_peers
+  do
+    if [ $interface == $(uci get wireguard.$i.ifname) ]; then
+      thispeer=$i
+      break;
+    fi
+  done
+
+  localtime=$(date +%s)
+  if [[ -n $thispeer ]]; then
+
+    remote=$(uci get wireguard.$thispeer.remote)
+    brokerport=$(uci get wireguard.$thispeer.broker_port)
+    localport=$(wg show $interface listen-port)
+
+    serverreply="$(nc_timeout 10 $remote $brokerport "{\"version\":1, \"pubkey\":\"$pubkey\", \"localport\":$localport }")"
```

Comment:
You can't possibly know the localport if the router is behind NAT, which is why I think we shouldn't send it at all.

Reply:
ok. On closer inspection the original intention cannot be reached with localport anyways. removed.